src/main/java/org/cyclonedx/util/ExtensibleTypesSerializer.java

```
@@ -0,0 +1,92 @@
+{
+  private final String XMLNS = "xmlns";
+
+  private final String DEFAULT_VALID_NAMESPACE = "http://www.w3.org/1999/xhtml";
+
+  public ExtensibleTypesSerializer() {
+    this(null);
+  }
+
+  public ExtensibleTypesSerializer(final Class t) {
+    super(t);
+  }
+
+  @Override
+  public void serialize(
+      final List<ExtensibleType> extList,
+      final JsonGenerator generator,
+      final SerializerProvider provider)
+      throws IOException
+  {
+    final ToXmlGenerator toXmlGenerator = (ToXmlGenerator) generator;
+    final XMLStreamWriter staxWriter = toXmlGenerator.getStaxWriter();
+    try {
+      if (extList != null && !extList.isEmpty()) {
+        for (ExtensibleType ext : extList) {
+          if (ext.getAttributes() != null && !ext.getAttributes().isEmpty()) {
+            Attribute xmlNS = ext.getAttributes().stream()
+                .filter(a -> a.getKey().contains(XMLNS))
+                .findAny()
+                .orElse(null);
+
+            if (xmlNS != null) {
+              staxWriter.writeStartElement(ext.getNamespace(), ext.getName(), xmlNS.getValue());
+              if (ext.getAttributes() != null && !ext.getAttributes().isEmpty()) {
+                for (Attribute attr : ext.getAttributes()) {
+                  staxWriter.writeAttribute(attr.getKey(), attr.getValue());
+                }
+              }
+            }
+          } else {
+            staxWriter.writeStartElement(ext.getNamespace(), ext.getName(), DEFAULT_VALID_NAMESPACE);
+          }
+
+          if (ext.getExtensibleTypes() != null && !ext.getExtensibleTypes().isEmpty()) {
+            serialize(ext.getExtensibleTypes(), generator, provider);
+          }
+          if (ext.getValue() != null) {
+            staxWriter.writeCharacters(ext.getValue());
+          }
+          staxWriter.writeEndElement();
+        }
+      }
+    }
+    catch (XMLStreamException e) {
+      e.printStackTrace();
```

I think it's better to throw the exception

---

Since this is an override, I'll have to throw it as `IOException`, do you see any issue with that?

---

In: b53fa28